Objects/stringlib/unicode_format.h

```diff
@@ -462,8 +462,17 @@ get_field_object(SubString *input, PyObject *args, PyObject *kwargs,
             tmp = getattr(obj, &name);
         else
             /* getitem lookup "[]" */
-            if (index == -1)
+            if (index == -1) {
+                if (PySequence_Check(obj)) {
+                    PyObject *str = SubString_new_object(&name);
```

Comment:
Can this be NULL?  Shouldn't you do Py_DECREF when you done with ``str``?

Reply:
Thank you, I'm new to CPython internals. Fixed.